packages/compartment-mapper/src/node-powers.js

```diff
@@ -30,9 +30,10 @@ const fakePathToFileURL = path => {
  * but `makeReadPowers` presents a type that requires `url`.
  *
  * @param {object} args
- * @param {typeof import('fs')} args.fs
- * @param {typeof import('url')} [args.url]
- * @param {typeof import('crypto')} [args.crypto]
+ * @param {import('./types.js').FsAPI} args.fs
+ * @param {import('./types.js').UrlAPI} [args.url]
+ * @param {import('./types.js').CryptoAPI} [args.crypto]
```

Comment:
I’m a tiny bit worried that, if we need more powers from these modules, reaching them going forward will constitute a breaking change. Is it possible to address your concerns without narrowing these types?

Reply:
I don't see it as possible to do without breaking things, no.  But currently, we're asking for way more than we need to.  

I'd even be amenable to ditching these objects entirely, and asking for the specific functions in isolation--i.e., instead of discrete `fs`, `url` and `crypto` objects, we'd ask for a bucket of functions matching what we need.

Reply:
I’m with you on principle of least authority but not aligned on resilience to change. This is a tool for attenuating the Node.js builtins down to the powers the compartment mapper already needs. The powers space is very narrowly scoped and any new powers have to be optional and have backward-compatible fallbacks. The attenuator however benefits from being able to reach for more powers as needed to upgrade gracefully. Last time we needed this, it was to reach for fs.realpath in order to canonicalize. I _hope_ the dependencies for compartment mapper are stable, but I’m not sure.